gui/default/syncthing/core/syncthingController.js

```diff
@@ -1732,16 +1732,16 @@ angular.module('syncthing.core')
         };
 
         $scope.setAllDevicesPause = function (pause) {
-            for (var id in $scope.devices) {
-                $scope.devices[id].paused = pause;
+            for (var id in $scope.otherDevices()) {
+                $scope.otherDevices()[id].paused = pause;
```

Comment:
```suggestion
                $scope.devices[id].paused = pause;
```

Same below

Reply:
I have tried changing the code like that, but I am getting tons of these JS errors.

```
TypeError: Cannot read property 'paused' of undefined
    at Scope.$scope.isAtleastOneDevicePausedStateSetTo (syncthingController.js:1744)
    at Object.$parseFunctionCall [as get] (angular.js:12493)
    at Scope.$digest (angular.js:14423)
    at Scope.$apply (angular.js:14694)
    at angular.js:14984
    at completeOutstandingRequest (angular.js:4959)
    at angular.js:5347
```

The modified code:

``` javascript
        $scope.setAllDevicesPause = function (pause) {
            for (var id in $scope.otherDevices()) {
                $scope.devices[id].paused = pause;
            };
            $scope.config.devices = deviceList($scope.otherDevices());
            $scope.saveConfig();
        }

        $scope.isAtleastOneDevicePausedStateSetTo = function (pause) {
            for (var id in $scope.otherDevices()) {
                if ($scope.devices[id].paused == pause) {
                    return true;
                }
            }

            return false
        }
```

Reply:
Ah, `otherDevices()` returns a list, i.e. the iterator variable is not a device-id, but a simple index. You can use `otherDevices.foreach(function (device) {...});` to iterate then.

Reply:
I have modified the code and it does work, but please check it and let me know what you think about it now.